apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/ValueConversion.cs

```diff
@@ -355,18 +343,72 @@ private static object ConvertToClrTypeImpl(Value wireValue, V1.Type spannerType,
                         return dictionary;
                     case Value.KindOneofCase.ListValue:
                         dictionary = new Hashtable();
-                        var i = 0;
-                        foreach (var listItemValue in wireValue.ListValue.Values)
+                        if (spannerType.Code == TypeCode.Struct)
                         {
-                            dictionary[i] = listItemValue.ConvertToClrType(spannerType.ArrayElementType);
-                            i++;
+                            for (int i = 0; i < spannerType.StructType.Fields.Count; i++)
+                            {
+                                dictionary[spannerType.StructType.Fields[i].Name] = wireValue
+                                    .ListValue.Values[i]
+                                    .ConvertToClrType(spannerType.StructType.Fields[i].Type);
+                            }
+                        }
+                        else
+                        {
+                            var i = 0;
+                            foreach (var listItemValue in wireValue.ListValue.Values)
```

Comment:
Why is a list being returned in a dictionary?
Shouldn't this be an actual `List` type of some sort?